devenv/main.py

```diff
@@ -67,47 +33,48 @@ def devenv(argv: Sequence[str], config_path: str) -> ExitCode:
     config = read_config(config_path)
 
     # Guessing temporary code root
-    coderoot = config.get("devenv", "coderoot", fallback=None) or (
+    code_root = config.get("devenv", "coderoot", fallback=None) or (
         os.path.abspath(f"{current_root}/..")
         if current_root
         else os.path.expanduser("~/code")
     )
 
-    if args.command == "bootstrap":
-        return bootstrap.main(config_path, coderoot, remainder)
+    modinfo_list = [
+        (module.__name__, module.module_info)
+        for module in [bootstrap, fetch, doctor, pin_gha, sync]
+        if hasattr(module, "module_info")
+    ]
 
-    if args.command == "fetch":
-        return fetch.main(coderoot, remainder)
+    parser = argparse.ArgumentParser()
+    subparser = parser.add_subparsers(
+        title=argparse.SUPPRESS, metavar="command", dest="module", required=True
+    )
 
-    if not args.nocoderoot and not os.getcwd().startswith(coderoot):
-        print(
-            f"You aren't in your code root ({coderoot})!\n"
-            "To ignore, use devenv --nocoderoot [COMMAND]\n"
-            f"To change your code root, you can edit {config_path}.\n"
+    for name, info in modinfo_list:
+        # Argparse stuff
+        child = subparser.add_parser(info.command, help=info.help)
+        child.add_argument(
+            "--xixax",
+            dest="_module_exec",
+            default=info.action,
+            required=False,
+            help=argparse.SUPPRESS,
         )
```

Comment:
this is clever -- but very hard to follow -- here's a simpler alternative that doesn't require a secret argument:

```python
modules = {
    mod.name: mod for mod in [...]
}
return modules[args.command].whatever(...)
```